src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesRunner.cs

```diff
@@ -0,0 +1,311 @@
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNameRequired));
+            }
+
+            // Check to see if we already have a registered source with the same name or source
+            var sourceList = args.SourceProvider.LoadPackageSources().ToList();
+
+            var source = args.SourceProvider.GetPackageSourceByName(args.Name);
+            if (source == null)
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNoMatchingSourcesFound, args.Name));
+            }
+
+            args.SourceProvider.RemovePackageSource(args.Name);
+            Console.WriteLine(string.Format(CultureInfo.CurrentCulture,
+                Strings.SourcesCommandSourceRemovedSuccessfully, args.Name));
+        }
+
+        private static void AddNewSource(SourcesArgs args)
+        {
+            if (string.IsNullOrEmpty(args.Name))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                                Strings.SourcesCommandNoMatchingSourcesFound));
+            }
+            if (string.Equals(args.Name, Strings.ReservedPackageNameAll))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                                Strings.SourcesCommandAllNameIsReserved));
+            }
+            if (string.IsNullOrEmpty(args.Source))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                                Strings.SourcesCommandSourceRequired));
+            }
+
+            // Make sure that the Source given is a valid one.
+            if (!PathValidator.IsValidSource(args.Source))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                                 Strings.SourcesCommandInvalidSource));
+            }
+
+            ValidateCredentials(args);
+
+            // Check to see if we already have a registered source with the same name or source
+            var existingSourceWithName = args.SourceProvider.GetPackageSourceByName(args.Name);
+            if (existingSourceWithName != null)
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                                Strings.SourcesCommandUniqueName));
+            }
+            var existingSourceWithSource = args.SourceProvider.GetPackageSourceBySource(args.Source);
+            if (existingSourceWithSource != null)
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                                Strings.SourcesCommandUniqueSource));
+            }
+
+            var newPackageSource = new Configuration.PackageSource(args.Source, args.Name);
+
+            if (!string.IsNullOrEmpty(args.Username))
+            {
+                var credentials = Configuration.PackageSourceCredential.FromUserInput(
```

Comment:
Same thing applies everywhere. 

NuGet.* should have a using include unless there's a type clash.

Reply:
fixing.